src/main/java/com/lafaspot/pop/client/PopClient.java

```diff
@@ -68,7 +66,36 @@ public PopClient(final int threads, @Nonnull final LogManager logManager) throws
             this.sslContext = SslContextBuilder.forClient().build();
             this.bootstrap.group(this.group);
             this.bootstrap.channel(NioSocketChannel.class);
-            this.bootstrap.option(ChannelOption.SO_KEEPALIVE, true); // (4)
+            this.bootstrap.option(ChannelOption.SO_KEEPALIVE, true);
+        } catch (final SSLException e) {
+            throw new PopException(PopException.Type.INTERNAL_FAILURE, e);
+        }
+    }
+
+    /**
+     * Constructor to create a new POP client with the system properties.
+     *
+     * @param properties the bootstrap properties
+     * @param threads number of threads to use
+     * @param logManager the log manager
+     * @throws PopException on failure
+     */
+    public PopClient(final int threads, @Nonnull final LogManager logManager, @Nonnull final Properties properties) throws PopException {
+
+        LogContext context = new SessionLogContext("PopClient");
+        this.logger = logManager.getLogger(context);
+        this.bootstrap = new Bootstrap();
+        this.group = new NioEventLoopGroup(threads);
+        try {
+            this.sslContext = SslContextBuilder.forClient().build();
+            this.bootstrap.group(this.group);
+            this.bootstrap.channel(NioSocketChannel.class);
+            this.bootstrap.option(ChannelOption.SO_KEEPALIVE, true);
+            final Iterator<Object> it = properties.keySet().iterator();
+            while (it.hasNext()) {
+                final ChannelOption obj = (ChannelOption) it.next();
+                this.bootstrap.option(obj, properties.get(obj));
```

Comment:
Have well defined property values that you look for and pass down.

Reply:
comparing the string from the properties keyset with a known set of keys.